x-pack/plugins/index_management/public/application/sections/home/index_list/index_table/index_table.js

```diff
@@ -107,6 +116,13 @@ export class IndexTable extends Component {
         this.setState({ filterError: e });
       }
     }
+
+    // Check if the we have the includeHidden query param
+    const { includeHidden } = parse((location && location.search) || '');
```

Comment:
I was thinking that we could remove `showHiddenIndices` from the store entirely and derive it from the URL inside of the `render` method. Then we don't need to worry about keeping the two in sync -- the URL is the source of truth for this state, as Seb mentioned.

```js
render() {
	// snip
    const { includeHidden } = parse((location && location.search) || '');
    const showHiddenIndices = includeHidden === 'true';
```

Reply:
I think I can see how that would work. We'd still need to hold a piece of local state to trigger a re-render when the toggle gets clicked however: `click toggle -> update url -> this.setState -> render`. To make it fully reliant on hash we can `window.addEventListener('hashchange', () => { this.setState })`- I am not convinced we should expect users to drive state this way though.

We still need to sync the URL on toggle, the difference being where (derived from store or derived from local state).

I think the current implementation is in keeping with the established pattern in this plugin so I would prefer to stick with it, but I can see how we can change this up when building bigger improvements in future.

Reply:
> I think the current implementation is in keeping with the established pattern in this plugin so I would prefer to stick with it

That sounds good to me. Also, I looked a little more deeply and it looks like there's a `getFilteredIndices` selector that depends upon `tableState.showHiddenIndices`, so removing it would be more involved than I thought (and possibly make the code more difficult to follow).

Reply:
If the URL is the source of truth, this is what we want to do

```js
import React, { useEffect } from 'react';
import { useLocation } from 'react-router';

function MyComponent() {
  const location = useLocation();

  useEffect(() => {
      console.log('route has been changed');
      ...your code
  },[location.pathname]);
}
```

Then the toggle is only responsible to update the URL.

Reply:
Ah, that is cool, I did not know about `useLocation`. Is there an equivalent approach for class components?

Reply:
This should do the same thing

```js
import {withRouter} from 'react-router-dom';

class MyComponent extends PureComponent {
  componentDidUpdate(prevProps) {
    if (this.props.location.pathname !== prevProps.location.pathname) {
      console.log('route has been changed');
      ...your code
    }
  }

  render() {
    return (
        ...
    );
  }
}

export default withRouter(MyComponent);
```